kyuubi-hive-jdbc/pom.xml

```diff
@@ -37,6 +37,57 @@
     </properties>
 
     <dependencies>
+
+        <dependency>
+            <groupId>org.apache.arrow</groupId>
+            <artifactId>arrow-vector</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.arrow</groupId>
+            <artifactId>arrow-memory-netty</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-annotations</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-core</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-databind</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>com.fasterxml.jackson.dataformat</groupId>
+            <artifactId>jackson-dataformat-yaml</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>com.fasterxml.jackson.datatype</groupId>
+            <artifactId>jackson-datatype-jsr310</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>com.fasterxml.jackson.jaxrs</groupId>
+            <artifactId>jackson-jaxrs-base</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>com.fasterxml.jackson.jaxrs</groupId>
+            <artifactId>jackson-jaxrs-json-provider</artifactId>
```

Comment:
why should we include jackson extensions? I saw arrow-vector depends on jackson-core, but is it a hard dependency or not? I checked the kyuubi-hive-jdbc-shaded, it's changed from 10mb to 17mb after this patch, better to keep it small

Reply:
Reduced pom.xml to minimum requirements, the shaded jar still has 16m size.

```
>ll -h kyuubi-hive-jdbc-shaded/target/kyuubi-hive-jdbc-shaded-1.7.0-SNAPSHOT.jar
-rw-r--r--  1 fchen  staff    16M 11 25 10:23 kyuubi-hive-jdbc-shaded/target/kyuubi-hive-jdbc-shaded-1.7.0-SNAPSHOT.jar
```

Reply:
if we split out the optional deps out, e.g. zk, arrow, the core jar should be small enough. it's something we are plan to do in the future.

Reply:
```
mvn dependency:tree -pl kyuubi-hive-jdbc | grep -v test
```

the output before this PR:

```
[INFO] org.apache.kyuubi:kyuubi-hive-jdbc:jar:1.7.0-SNAPSHOT
[INFO] +- org.apache.thrift:libfb303:jar:0.9.3:compile
[INFO] +- org.apache.thrift:libthrift:jar:0.9.3:compile
[INFO] +- org.apache.hive:hive-service-rpc:jar:3.1.3:compile
[INFO] +- commons-codec:commons-codec:jar:1.15:compile
[INFO] +- org.apache.commons:commons-lang3:jar:3.10:compile
[INFO] +- com.google.guava:guava:jar:30.1-jre:compile
[INFO] |  \- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO] +- com.google.code.findbugs:jsr305:jar:3.0.2:provided
[INFO] +- org.apache.curator:curator-framework:jar:2.12.0:compile
[INFO] +- org.apache.curator:curator-client:jar:2.12.0:compile
[INFO] +- org.apache.httpcomponents:httpclient:jar:4.5.13:compile
[INFO] |  \- org.apache.httpcomponents:httpcore:jar:4.4.15:compile
[INFO] +- org.apache.zookeeper:zookeeper:jar:3.4.14:compile
[INFO] +- org.slf4j:jcl-over-slf4j:jar:1.7.35:compile
[INFO] +- org.slf4j:slf4j-api:jar:1.7.35:compile
```

the output after this PR:

```diff
 [INFO] org.apache.kyuubi:kyuubi-hive-jdbc:jar:1.7.0-SNAPSHOT
+[INFO] +- org.apache.arrow:arrow-vector:jar:10.0.0:compile
+[INFO] |  +- org.apache.arrow:arrow-format:jar:10.0.0:compile
+[INFO] |  +- org.apache.arrow:arrow-memory-core:jar:10.0.0:compile
+[INFO] |  \- com.google.flatbuffers:flatbuffers-java:jar:1.12.0:compile
+[INFO] +- org.apache.arrow:arrow-memory-netty:jar:10.0.0:compile
+[INFO] |  +- io.netty:netty-buffer:jar:4.1.73.Final:compile
+[INFO] |  \- io.netty:netty-common:jar:4.1.73.Final:compile
+[INFO] +- com.fasterxml.jackson.core:jackson-annotations:jar:2.13.4:compile
+[INFO] +- com.fasterxml.jackson.core:jackson-core:jar:2.13.4:compile
+[INFO] +- com.fasterxml.jackson.core:jackson-databind:jar:2.13.4:compile
+[INFO] +- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.13.4:compile
 [INFO] +- org.apache.thrift:libfb303:jar:0.9.3:compile
 [INFO] +- org.apache.thrift:libthrift:jar:0.9.3:compile
 [INFO] +- org.apache.hive:hive-service-rpc:jar:3.1.3:compile
 [INFO] +- commons-codec:commons-codec:jar:1.15:compile
 [INFO] +- org.apache.commons:commons-lang3:jar:3.10:compile
 [INFO] +- com.google.guava:guava:jar:30.1-jre:compile
 [INFO] |  \- com.google.guava:failureaccess:jar:1.0.1:compile
 [INFO] +- com.google.code.findbugs:jsr305:jar:3.0.2:provided
 [INFO] +- org.apache.curator:curator-framework:jar:2.12.0:compile
 [INFO] +- org.apache.curator:curator-client:jar:2.12.0:compile
 [INFO] +- org.apache.httpcomponents:httpclient:jar:4.5.13:compile
 [INFO] |  \- org.apache.httpcomponents:httpcore:jar:4.4.15:compile
 [INFO] +- org.apache.zookeeper:zookeeper:jar:3.4.14:compile
 [INFO] +- org.slf4j:jcl-over-slf4j:jar:1.7.35:compile
 [INFO] +- org.slf4j:slf4j-api:jar:1.7.35:compile
```